### PR TITLE
[Reland]: Fix `StretchingOverscrollIndicator` clipping and add `clipBehavior` parameter

### PR DESCRIPTION
**Pull request by TahaTesser**
Reland  https://github.com/flutter/flutter/pull/105303 with minor tweaks.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 1]
```diff
@@ -1326,6 +1337,13 @@ class ScrollableDetails {
   /// This can be used by [ScrollBehavior] to apply a [Scrollbar] to the associated
   /// [Scrollable].
   final ScrollController controller;
+
+  /// {@macro flutter.material.Material.clipBehavior}
+  ///
+  /// This can be used by [MaterialScrollBehavior] to clip [StretchingOverscrollIndicator].
+  ///
+  /// Cannot be null.
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply packages/flutter/lib/src/widgets/scrollable.dart:1345]: Forget to remove this, thanks!

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 2]
```diff
@@ -1312,11 +1323,12 @@ class ScrollableDetails {
   const ScrollableDetails({
     required this.direction,
     required this.controller,
+    this.clipBehavior,
   });
 
   /// The direction in which this widget scrolls.
   ///
-  /// Cannot be null.
+  /// Defaults to null.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/scrollable.dart:1331]: Oh I think the wrong docs were updated. Just a mix up nit. 🙃

File: packages/flutter/lib/src/widgets/scrollable.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Piinks) [reply packages/flutter/lib/src/widgets/scrollable.dart:1345]: See above.

**Review by Piinks — APPROVED** [review]
LGTM!

**Pull request merged**